server/routes/article.js

```diff
@@ -1,26 +1,30 @@
 import { Router } from 'express';
 import ArticleController from '../controllers/ArticleController';
-import userAuthenticate from '../middlewares/isLoggedIn';
+import isLoggedIn from '../middlewares/isLoggedIn';
 import ArticleValidation from '../middlewares/validations/ArticleValidation';
 import CommentValidation from '../middlewares/validations/CommentValidation';
 import CommentController from '../controllers/CommentController';
+import RatingController from '../controllers/RatingController';
 
 const articleRouter = Router();
 
-articleRouter.post('/', userAuthenticate, ArticleValidation.validateCreateArticle, ArticleController.createArticle);
+articleRouter.post('/', isLoggedIn, ArticleValidation.validateCreateArticle, ArticleController.createArticle);
 
-articleRouter.put('/:slug', userAuthenticate, ArticleValidation.validateUpdateArticle, ArticleController.updateArticle);
+articleRouter.put('/:slug', isLoggedIn, ArticleValidation.validateUpdateArticle, ArticleController.updateArticle);
 
-articleRouter.delete('/:slug', userAuthenticate, ArticleController.removeArticle);
+articleRouter.delete('/:slug', isLoggedIn, ArticleController.removeArticle);
 
-articleRouter.post('/:slug/comments', userAuthenticate, CommentValidation.validateComment, CommentController.createComment);
+articleRouter.post('/:slug/comments', isLoggedIn, CommentValidation.validateComment, CommentController.createComment);
 
-articleRouter.get('/:slug/comments', userAuthenticate, CommentController.getComments);
+articleRouter.get('/:slug/comments', isLoggedIn, CommentController.getComments);
 
-articleRouter.get('/:slug/comments/:id', userAuthenticate, CommentController.getComment);
+articleRouter.get('/:slug/comments/:id', isLoggedIn, CommentController.getComment);
 
-articleRouter.put('/:slug/comments/:id', userAuthenticate, CommentValidation.validateComment, CommentController.updateComment);
+articleRouter.put('/:slug/comments/:id', isLoggedIn, CommentValidation.validateComment, CommentController.updateComment);
+
+articleRouter.delete('/:slug/comments/:id', isLoggedIn, CommentController.deleteComment);
+
+articleRouter.post('/:author/:slug/:rating', isLoggedIn, RatingController.rateArticle);
```

Comment:
I'm a bit confused by this route, @Veraclins. Would you please describe how this route works? I can see both `author` and `slug`. Thanks.

Reply:
@madeofhuman thanks for your review and nice words. Regarding the route, the `author` refers to the username of the author of the article. As will become evident when @chukwuemekachm adds the routes for deleting and updating an article, all operations involving single articles requires that we know the author of the article. In the case of deleting or updating we must ensure that the person trying to carry out the operation is actually the author of the article while for rating, we must ensure the author does not rate his or her own article.
All these require that we make a comparison between who is logged in and the author of the given article. With the author's username in the URL, all we need to do is get it as a request parameter and make the necessary comparison. 
The alternative is that we have the URL without the author's username and then make a db call to get the author of the article before making the comparison. This means that every request to any of those endpoints will make one extra db call. If we add this to those operations that already involve multiple db calls, we may end up having up to three db calls for a single operation. Hence our decision to include the authors's username in the URL especially since that is how it is done on medium.com

Reply:
Thanks for clearing that up. Good work, man.